### ISSUE TITLE
requireAuthOnInitialize : false doesnt seem to work

### ISSUE THREAD

**Issue by darthgus** [survirtual/vue-adal#14]
Was trying to play around and turn off auth on initialize, so I could set it per route, but setting the flag to false doesnt seem to prevent it still going off to login when going to the site for the first time.

requireAuthOnInitialize : false

is there anything else I need to do?

**Comment by yeya** [comment]
Can you send code example? Make sure you are not initialize the same route that require auth

**Comment by chrishannaby** [comment]
I've noticed this behaviour too. It seems to be when you configure an axios client interceptor in the way demonstrated in the sample you will be redirected to login immediately even if requireAuthOnInitialize is set to false. Is there an alternative way of setting up the interceptor after the user has manually logged in?

**Comment by survirtual** [comment]
I believe this behavior is due to caching.  Have you tried with a clear browser cache?

**Comment by dan-sls** [comment]
I don't think this is caching, as I'm testing from a fresh Chrome incognito window. I'm seeing the same behavior. I want to make one of my routes not require a login.

**Comment by yeya** [comment]
Do yourself a favor, move to msal.
adal is a nightmare...

**Comment by tomjdickson** [comment]
I am also experiencing this issue. Caching does not seem to be the problem.

**Comment by vanna80** [comment]
I think that this behavior is because of calling some function which requires Auth. 
Specifically in sample/example code: In Home.vue, there is:
      let userInfo = await this.getUserInfo()
Calling this API requires Auth. So the page is implicitly redirected to login page despite the requireAuthOnInitialize set to false.
Just remove the line mentioned above and page won't be redirected.

**Comment by AngelReyesEspinal** [comment]
What I did:
isAnonymousRoute is a function to validate if the current route requires Auth, ((you can made your own logic here))

for example:
    path: '/about',
    name: 'About',
    component: () => import(/* webpackChunkName: "about" */ '../views/About.vue'),
    meta: {
      requireAuth: false
    }
 
then in the vue-adal config:

import { AuthenticationContext } from 'vue-adal' // This will be populated with a valid context after initialization

AuthenticationContext.acquireToken(resource, (err, token) => {
    if (err) {
      let errCode = err.split(':')[0]
switch (errCode) {
                case 'AADSTS50058': // Need to prompt for user sign in
                    if (!isAnonymosRoute) 
                        AuthenticationContext.login();
                    break
                case 'AADSTS65001': // Token is invalid; grab a new one
                    AuthenticationContext.acquireTokenRedirect(config.clientid);
                    break
                case 'AADSTS16000': // No Access
                default: // Need a pop-up forcing a login
                    if (!isAnonymosRoute) 
                        AuthenticationContext.login();
                    break
            }
 return
    }
    const headers = {
      'Authorization': `BEARER ${token}`
    }
  })

**Comment by peterfarrell** [comment]
@AngelReyesEspinal How did you actually wire in the AuthenticationContext.acquireToken in your vue-adal config? Could you share that code?
